src/server.ts

```diff
@@ -10,33 +10,61 @@ import type { GraphQLContext } from './context';
 
 let LOG_LEVEL = (process.env.LOG_LEVEL as LogLevel) || 'info';
 
-export function buildServer(context: GraphQLContext) {
+function initJaegerProvider() {
+  let provider = undefined;
+  if (process.env.ENABLE_JAEGER) {
+    provider = buildProvider();
+    if (!process.env.JAEGER_ENDPOINT) {
+      throw new Error(
+        'Jaeger endpoint not found. Please ensure that the Jaeger endpoint is properly configured and available.'
+      );
+    }
+    if (!process.env.JAEGER_SERVICE_NAME) {
+      throw new Error(
+        'Jaeger service name not found. Please ensure that the Jaeger service name is properly configured and available.'
+      );
+    }
+  }
+  return provider;
+}
+
+function buildPlugins() {
   let plugins = [];
+
   plugins.push(useGraphQlJit());
-  plugins.push(
-    useOpenTelemetry(
-      {
-        resolvers: false, // Tracks resolvers calls, and tracks resolvers thrown errors
-        variables: true, // Includes the operation variables values as part of the metadata collected
-        result: true, // Includes execution result object as part of the metadata collected
-      },
-      process.env.ENABLE_JAEGER ? buildProvider() : undefined
-    )
-  );
-  if (process.env.ENABLE_INTROSPECTION !== 'true')
+
+  if (process.env.ENABLE_LOGGING === 'true') {
+    let provider = initJaegerProvider();
+    plugins.push(
+      useOpenTelemetry(
+        {
+          resolvers: false, // Tracks resolvers calls, and tracks resolvers thrown errors
+          variables: true, // Includes the operation variables values as part of the metadata collected
+          result: true, // Includes execution result object as part of the metadata collected
+        },
+        provider
+      )
+    );
+  }
+
+  if (!process.env.ENABLE_INTROSPECTION)
     plugins.push(useDisableIntrospection());
 
+  return plugins;
+}
+
+export function buildServer(context: GraphQLContext) {
   const yoga = createYoga<GraphQLContext>({
     schema,
     logging: LOG_LEVEL,
     graphqlEndpoint: '/',
     landingPage: false,
     healthCheckEndpoint: '/healthcheck',
```

Comment:
Not as part of this particular PR but `/healthcheck` endpoint doesn't respond to used on `HTTP GET` with text ort json, only status code with empty body.

Reply:
Is a status code enough for the health check? I don't see much value in adding text/json if a 200 status code is returned.

Reply:
I don't really see value of that endpoint at all, if service won't operate then users will hit whatever error without it :)

Reply:
Doesn't it fulfill the purpose if it returns 200? If the server weren't live, it wouldn't return anything, right?

Reply:
That is the thing, any other endpoint will return error as well so it doesn't bring much value but we can have it anyway since it is is there for free.